src/test/java/com/shapesecurity/shift/es2016/test262/Test262.java

```diff
@@ -0,0 +1,338 @@
+			"src/test/resources/test262/test/language/expressions/exponentiation/exp-operator-syntax-error-negate-unary-expression-base.js",
+			"src/test/resources/test262/test/language/expressions/exponentiation/exp-operator-syntax-error-plus-unary-expression-base.js",
+			"src/test/resources/test262/test/language/expressions/exponentiation/exp-operator-syntax-error-bitnot-unary-expression-base.js",
+
+			// yield not implemented
+			"src/test/resources/test262/test/language/expressions/assignment/dstr-obj-id-identifier-yield-expr.js",
+			"src/test/resources/test262/test/language/statements/for-in/dstr-obj-id-identifier-yield-expr.js",
+			"src/test/resources/test262/test/language/statements/for-of/dstr-obj-id-identifier-yield-expr.js",
+
+			// shift-java#208
+			"src/test/resources/test262/test/language/global-code/decl-lex-restricted-global.js"
+	));
+
+	private static final HashSet<String> skip = new HashSet<>(Arrays.asList(
+			// async/await not implemented
+			"src/test/resources/test262/test/language/expressions/async-arrow-function",
+			"src/test/resources/test262/test/language/expressions/async-function",
+			"src/test/resources/test262/test/language/statements/async-function",
+			"src/test/resources/test262/test/built-ins/AsyncFunction",
+			"src/test/resources/test262/test/language/expressions/await",
+			// no regex acceptor
+			"src/test/resources/test262/test/language/literals/regexp",
+			// generators not implemented
+			"src/test/resources/test262/test/language/expressions/generators",
+			"src/test/resources/test262/test/language/statements/generators",
+
+			// no module early errors
+			"src/test/resources/test262/test/language/module-code"
+	));
+
+	private static final String testsDir = "src/test/resources/test262/test/";
+
+	private enum Test262Negative {
+		PARSE, EARLY, EXECUTE, NONE
+	}
+
+	private static final class Test262Info {
+		@Nonnull
+		public final String name;
+		public final Test262Negative negative;
+		public final boolean noStrict;
+		public final boolean onlyStrict;
+		public final boolean async;
+		public final boolean module;
+
+		public Test262Info(@Nonnull String name, @Nonnull Test262Negative negative, boolean noStrict, boolean onlyStrict, boolean async, boolean module) {
+			this.name = name;
+			this.negative = negative;
+			this.noStrict = noStrict;
+			this.onlyStrict = onlyStrict;
+			this.async = async;
+			this.module = module;
+		}
+	}
+
+	@Nullable
+	private static Test262Info extractTest262Info(@Nonnull String path, @Nonnull String source) {
+		// extract comment block
+		int test262CommentBegin = source.indexOf("/*---");
+		if (test262CommentBegin < 0) {
+			return null;
+		}
+		test262CommentBegin += 5;
+		int test262CommentEnd = source.indexOf("---*/", test262CommentBegin);
+		if (test262CommentEnd < 0) {
+			return null;
+		}
+
+		// extract categories -- accounting for various authors indentations.
+		Scanner scanner = new Scanner(source.substring(test262CommentBegin, test262CommentEnd));
```

Comment:
I think we probably just want to depend on a yaml parser.

Reply:
Added snakeyaml